Tests/Packet++Test/Tests/NtpTests.cpp

```diff
@@ -0,0 +1,158 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "EndianPortable.h"
+#include "Packet.h"
+#include "EthLayer.h"
+#include "IPv4Layer.h"
+#include "UdpLayer.h"
+#include "NtpLayer.h"
+#include "SystemUtils.h"
+
+PTF_TEST_CASE(NtpMethodsTests)
+{
+
+    double val = 12345.125;
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertFromTimestampFormat(pcpp::NtpLayer::convertToTimestampFormat(val)), val);
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertFromShortFormat(pcpp::NtpLayer::convertToShortFormat(val)), val);
+
+    // First check the epoch is correct
+#if defined(WIN32) || defined(WINx64)
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertToIsoFormat(0.0), std::string("1970-01-01T00:00:00.000000000Z"));
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertToIsoFormat(uint64_t(0)), std::string("1970-01-01T00:00:00.000000000Z"));
+#else
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertToIsoFormat(0.0), std::string("1970-01-01T00:00:00.000000000Z"));
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertToIsoFormat(uint64_t(0)), std::string("1900-01-01T00:00:00.000000000Z"));
+#endif
+
+
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertToIsoFormat(1642879410.0), "2022-01-22T19:23:30.000000000Z");
+    PTF_ASSERT_EQUAL(pcpp::NtpLayer::convertToIsoFormat(pcpp::NtpLayer::convertToTimestampFormat(1642879410.0)), "2022-01-22T19:23:30.000000000Z");
+
+} // NtpMethodsTests
+
+PTF_TEST_CASE(NtpParsingV3Tests)
+{
+
+    timeval time;
+    gettimeofday(&time, NULL);
+
+    READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/ntpv3.dat");
+
+    pcpp::Packet ntpPacket(&rawPacket1);
+    pcpp::NtpLayer *ntpLayer = ntpPacket.getLayerOfType<pcpp::NtpLayer>();
+
+    PTF_ASSERT_NOT_NULL(ntpLayer);
+    PTF_ASSERT_EQUAL(ntpLayer->getVersion(), 3);
+    PTF_ASSERT_EQUAL(ntpLayer->getLeapIndicator(), pcpp::NoWarning);
+    PTF_ASSERT_EQUAL(ntpLayer->getMode(), pcpp::Server);
+    PTF_ASSERT_EQUAL(ntpLayer->getStratum(), 0);
+    PTF_ASSERT_EQUAL(ntpLayer->getPollInterval(), 4);
+    PTF_ASSERT_EQUAL(ntpLayer->getPrecision(), int8_t(-6));
+    // NTPv3 pcap is a bit useless, too many zeros but these fields same with v4
+    PTF_ASSERT_EQUAL(ntpLayer->getRootDelay(), 0);
+    PTF_ASSERT_EQUAL(ntpLayer->getRootDispersion(), 0);
+    PTF_ASSERT_EQUAL(ntpLayer->getReferenceIdentifier(), 0);
+    PTF_ASSERT_EQUAL(ntpLayer->getReferenceTimestamp(), 0);
+    PTF_ASSERT_EQUAL(ntpLayer->getOriginateTimestamp(), 0);
+    PTF_ASSERT_EQUAL(ntpLayer->getReceiveTimestamp(), be64toh(0xd94f4f1100000000));
+    PTF_ASSERT_EQUAL(ntpLayer->getTransmitTimestamp(), be64toh(0xd94f4f1100000000));
+
+} // NtpParsingV3Tests
+
+PTF_TEST_CASE(NtpParsingV4Tests)
```

Comment:
If you want to find NTP packet that have more diverse information you can see here: https://weberblog.net/wp-content/uploads/2018/08/NTP-IPv6-with-Auth.zip